azurerm/internal/services/mssqlvm/tests/resource_arm_mssql_virtual_machine_test.go

```diff
@@ -0,0 +1,317 @@
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMMsSqlVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMMsSqlVirtualMachine_withStorage(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMsSqlVirtualMachineExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "sql_server_license_type", "PAYG"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMMsSqlVirtualMachineExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Sql Virtual Machine not found: %s", resourceName)
+		}
+
+		resourceGroupName := rs.Primary.Attributes["resource_group_name"]
+		id, _ := azure.ParseAzureResourceID(rs.Primary.Attributes["virtual_machine_resource_id"])
+		name := id.Path["virtualMachines"]
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).MSSQLVM.SQLVirtualMachinesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if resp, err := client.Get(ctx, resourceGroupName, name, ""); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q) does not exist", name, resourceGroupName)
+			}
+			return fmt.Errorf("Bad: Get on sqlVirtualMachinesClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMMsSqlVirtualMachineDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).MSSQLVM.SQLVirtualMachinesClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_sql_virtual_machine" {
+			continue
+		}
+
+		resourceGroupName := rs.Primary.Attributes["resource_group_name"]
+		id, _ := azure.ParseAzureResourceID(rs.Primary.Attributes["virtual_machine_resource_id"])
+		name := id.Path["virtualMachines"]
+
+		if resp, err := client.Get(ctx, resourceGroupName, name, ""); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on sqlVirtualMachinesClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMVirtualMachine_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "accsa%d"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_kind             = "StorageV2"
+  account_tier             = "Premium"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctvn-%d"
```

Comment:
We should always start with acctest, and have capitals in the name if the API allows it:
```suggestion
  name                = "acctest-VN-%d"
```